instrumentation/grpctrace/interceptor.go

```diff
@@ -427,11 +425,21 @@ func StreamServerInterceptor(tracer trace.Tracer) grpc.StreamServerInterceptor {
 	}
 }
 
-func peerInfoFromTarget(target string) []kv.KeyValue {
-	host, port, err := net.SplitHostPort(target)
+// spanInfo returns a span name and all appropriate attributes from the gRPC
+// method and peer address.
+func spanInfo(fullMethod, peerAddress string) (string, []kv.KeyValue) {
+	attrs := []kv.KeyValue{standard.RPCSystemGRPC}
+	name, mAttrs := parseFullMethod(fullMethod)
+	attrs = append(attrs, mAttrs...)
+	attrs = append(attrs, peerAttr(peerAddress)...)
+	return name, attrs
+}
 
+// peerAttr returns attributes about the peer address.
+func peerAttr(addr string) []kv.KeyValue {
+	host, port, err := net.SplitHostPort(addr)
 	if err != nil {
-		return []kv.KeyValue{}
+		return []kv.KeyValue(nil)
```

Comment:
Do we consistently return a typecast `nil` in the codebase? 

I personally like it, but for example, I note that in this same PR, `parseFullMethod()` at https://github.com/open-telemetry/opentelemetry-go/pull/922/files#diff-9ddb8ec663434d0cbad4f88de277a649R467 uses `return name, nil` when the second result parameter is of type `[]kv.KeyValue`.

Reply:
Good question. Functionally they are equivalent (any caller will a correctly typed (`[]kv.KeyValue`) `nil` value), so I guess it's going to be more about the readability in the code base and it succinctness.

I'm fine either way, but feel we should try to be consistent (at least in the same file). I'm happy to unify this whichever way you prefer.

Reply:
I think the only other place in this file is the `parseFullMethod()` occurrence, so let's just make that consistent with this typecast.

Reply:
Unified on `[]kv.KeyValue(nil)`. LMK if you think otherwise.